### ISSUE TITLE
Missing file extension when migrate from 2.18 to 2.19.1

### ISSUE THREAD

**Issue by jonathantribouharet** [shrinerb/shrine#399]
### Brief Description
Hi, I've just update from 2.18 to 2.19.1 and the file extension are now missing.
The file are PDF generated server side, not uploaded.
I think the problem is because I use `StringIO`.
No problem with uploaded files.

The metadata contain:
```
{"filename"=>nil, "size"=>18238, "mime_type"=>"application/pdf"}
```

### Simplest self-contained example code to demonstrate issue

```
require 'shrine'

Shrine.plugin :activerecord
Shrine.plugin :determine_mime_type
Shrine.plugin :infer_extension, inferrer: :mini_mime, force: true
Shrine.plugin :logging
Shrine.plugin :pretty_location
Shrine.plugin :validation_helpers

Shrine.storages = {
  cache: Shrine::Storage::FileSystem.new(Dir.tmpdir),
  tmp: Shrine::Storage::FileSystem.new('public', prefix: 'uploads/tmp'),
  store: Shrine::Storage::FileSystem.new('public', prefix: 'uploads'),
}

class DocumentUploader < Shrine

  Attacher.validate do
    validate_max_size 15 * 1024 ** 2, message: -> (max) { [:file_max_size, max: max] }
  end

end

class Order < ApplicationRecord

  include ::DocumentUploader::Attachment.new(:document)

end

order = ::Order.find(order_id)
content = File.read('example.pdf')
order.document = StringIO.new(content)

order.save
```

### System configuration
2.6.3

**Shrine version**:
2.19.1

**Comment by janko** [comment]
Thanks for the report, I know which change broke it. I will push a fix later today and push a new version out.

**Comment by jonathantribouharet** [comment]
Thanks.
I think I found a solution just by changing the order of the plugins by:

```
Shrine.plugin :activerecord
Shrine.plugin :pretty_location
Shrine.plugin :determine_mime_type
Shrine.plugin :infer_extension, inferrer: :mini_mime, force: true
Shrine.plugin :logging, logger: Rails.logger
Shrine.plugin :validation_helpers
```

**Comment by janko** [comment]
Yes, the bug happens when `pretty_location` plugin is loaded after `infer_extension` plugin.

**Comment by janko** [comment]
Fixed in https://github.com/shrinerb/shrine/commit/9f6b72b6b9404b539aa7fade6fe6e5ca748d3b4c (included in the 2.19.2 release)
